### ISSUE TITLE
EMV TLV(Tag Length Value) decoder/encoder. aka. Parse Field 55(DE55)

### ISSUE THREAD

**Issue by wadearnold** [moov-io/iso8583#100]
Create the ability to read and write EMV TLV data elements. This should be its own package and used for field 55 in iso8583. 

EMV Chip and Tag - EMV TLV - ISO8583 Data Element Field 55 - iso7816-4

"EMV Tags" are usually mile-long alpha-numeric strings in a BER-TLV format. "TLV" or as it might be referred to as "SIMPLE-TLV" stands for "Tag Length Value". BER-TLV is an "expanded" version of "TLV" that allows supplying tags with variable length, include one "TLV" inside another and provides few extra capabilities (for more details please refer to [stackoverflow post](https://stackoverflow.com/questions/18853800/simple-tlv-vs-ber-tlv)). 

Complete list of Field 55:ICC data
https://developer-eu.elavon.com/docs/eisop/field-descriptions/field-55-icc-data

EMV Tag Search
https://emvlab.org/emvtags/all/
https://neapay.com/online-tools/emv-tags-list.html

This application decodes EMV records encoded in tag-length-value (TLV) format
https://emvlab.org/tlvutils/
https://neapay.com/online-tools/emv-tlv-decoder.html

Complete list of Field 55:ICC data
https://developer-eu.elavon.com/docs/eisop/field-descriptions/field-55-icc-data

TLV also knows as ANS.1 BER (Basic Encoding Rules) from X.208
https://github.com/mercari/go-emv-code
https://github.com/binaryfoo/emv-bertlv
https://golang.org/src/encoding/asn1/asn1.go
https://github.com/skythen/bertlv
https://github.com/hexdigest/bertlv
https://stackoverflow.com/questions/18853800/simple-tlv-vs-ber-tlv

**Comment by krishishah** [comment]
To implement this successfully, I think we may need to make a breaking change to the `field.Spec` object:

```
type Spec struct {
	Length int
	// Only applicable Composite and other bespoke field types.
	IDLength    int
	Description string
	Enc         encoding.Encoder
	Pref        prefix.Prefixer
	Pad         padding.Padder
	Fields      map[int]Field
}
```

Notably, the `Fields` field may need to switch from a `map[int]Field` type to a `map[string]Field` type. This is because TLVs are keyed by hex rather than int (used by DEs, SFs and SEs). I considered making this change when implementing `field.Composite`, however didn't want to prematurely optimise a solution before we tried to implement it.

**Comment by wadearnold** [comment]
This has been closed by  #121
